controllers/storageconfig_controller.go

```diff
@@ -53,7 +54,10 @@ func newStorageSecret(dataConnectionSecretsList *corev1.SecretList) *corev1.Secr
 		dataConnectionElement["access_key_id"] = string(secret.Data["AWS_ACCESS_KEY_ID"])
 		dataConnectionElement["secret_access_key"] = string(secret.Data["AWS_SECRET_ACCESS_KEY"])
 		dataConnectionElement["endpoint_url"] = string(secret.Data["AWS_S3_ENDPOINT"])
+		// We add both "default_bucket" and "bucket" because kserve and modelmesh expect different keys for the same value
+		// Once upstream has reconciled to one common key, we should remove the other one.
 		dataConnectionElement["default_bucket"] = string(secret.Data["AWS_S3_BUCKET"])
+		dataConnectionElement["bucket"] = string(secret.Data["AWS_S3_BUCKET"])
```

Comment:
Is there any unit test for this?

Reply:
covered by the changes to `controllers/testdata/secrets/storageconfig-encoded.yaml` where the new encoded value expects the new `bucket` key